hilti/runtime/CMakeLists.txt

```diff
@@ -5,6 +5,17 @@ set(AUTOGEN_CC "${PROJECT_BINARY_DIR}/hilti/runtime/src/autogen")
 file(MAKE_DIRECTORY "${AUTOGEN_H}" "${AUTOGEN_CC}")
 file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/bin" "${PROJECT_BINARY_DIR}/lib")
 
+# Create a version string that's a valid C identifier.
+set(SPICY_VERSION_C_IDENT "${SPICY_VERSION}")
+
+if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+    # Distinguish between release and debug builds.
+    set(SPICY_VERSION_C_IDENT "${SPICY_VERSION_C_IDENT}_debug")
+endif ()
+
+string(REGEX REPLACE "-[0-9]*$" "_git" SPICY_VERSION_C_IDENT "${SPICY_VERSION_C_IDENT}")
+string(REGEX REPLACE "[^a-zA-Z0-9_\$]" "_" SPICY_VERSION_C_IDENT "${SPICY_VERSION_C_IDENT}")
+
```

Comment:
This computes a name just like e.g., Zeek does, and the name will include dev tags, e.g., `1_6_0_dev_71_debug`. Since this ends up in `libhilti.h` which is included in every JIT run this will mean that with such an identifier ccache during JIT will only find a hit if running on the exact same commit again.

WDYT about using a more stable string instead e.g., one derived from `SPICY_VERSION_NUMBER`? We already can make little promises to users about running from different dev snapshots, but this would allow us to cache more information.

Reply:
Makes sense, I'll change.